R/diet.R

```diff
@@ -0,0 +1,152 @@
+    idx_sp <- (no_w_full - no_w + 1):no_w_full
+
+    if (length(params@ft_pred_kernel_e) == 1) {
+        ae <- matrix(params@pred_kernel[, , idx_sp, drop = FALSE],
+                     ncol = no_w) %*% t(sweep(n, 2, params@w * params@dw,
+                                              "*"))
+        diet[, , 1:no_sp] <- ae
+        # just need an example before editing this line, usually the if condition is not fulfilled
+        diet[, , no_sp + 1] <- rowSums(sweep(params@pred_kernel, 3,
+                                             params@dw_full * params@w_full * n_pp, "*"), dims = 2)
+    }
+    else {
+        prey <- matrix(0, nrow = no_sp + no_other, ncol = no_w_full)
+        prey[1:no_sp, idx_sp] <- sweep(n, 2, params@w * params@dw, "*")
+        prey[(no_sp + 1):(no_sp + no_other), ] <- sweep(n_other$MR,2,
+                                                        params@w_full * params@dw_full, "*")
+        ft <- array(rep(params@ft_pred_kernel_e, times = no_sp + no_other) *
+                        rep(mvfft(t(prey)), each = no_sp), dim = c(no_sp, no_w_full, no_sp + no_other))
+        ft <- matrix(aperm(ft, c(2, 1, 3)), nrow = no_w_full)
+        ae <- array(Re(mvfft(ft, inverse = TRUE)/no_w_full),
+                    dim = c(no_w_full, no_sp, no_sp + no_other))
+        ae <- ae[idx_sp, , , drop = FALSE]
+        ae <- aperm(ae, c(2, 1, 3))
+        ae[ae < 1e-18] <- 0
+        diet[, , 1:(no_sp + no_other)] <- ae
+    }
+
+    inter <- cbind(params@interaction,resource_interaction(params))
+    diet[, , 1:(no_sp+no_other)] <- sweep(sweep(diet[, , 1:(no_sp+no_other), drop = FALSE],
+                                                c(1, 3), inter, "*"), c(1, 2), params@search_vol, "*")
+
+    f <- getFeedingLevel(object = params, n = n, n_other = n_other)
+    fish_mask <- n > 0
+    diet <- sweep(diet, c(1, 2), (1 - f) * fish_mask, "*")
+    if (proportion) {
+        total <- rowSums(diet, dims = 2)
+        diet <- sweep(diet, c(1, 2), total, "/")
+        diet[is.nan(diet)] <- 0
+    }
+    return(diet)
+}
+
+
+#' Plot diet, resolved by prey species, as function of predator at size.
+#'
+#' `r lifecycle::badge("experimental")`
+#' Plots the proportions with which each prey species contributes to the total
+#' biomass consumed by the specified predator species, as a function of the
+#' predator's size. These proportions are obtained with `getDietMR()`.
+#'
+#' Prey species that contribute less than 1 permille to the diet are suppressed
+#' in the plot.
+#'
+#' @inheritParams plotSpectra
+#' @param species The name of the predator species for which to plot the diet.
+#'
+#' @return A ggplot2 object, unless `return_data = TRUE`, in which case a data
+#'   frame with the three variables 'w', 'Proportion', 'Prey' is returned.
+#' @export
+#' @seealso [getDietMR()]
+#' @family plotting functions
+
+
+plotDietMR <- function (sim, species = NULL, time_range, xlim = c(1, NA), returnData = F)
```

Comment:
This should accept also a MizerParams object, like the ordinary `plotDiet()`. The `returnData` argument should be renamed to `return_data` as in `plotDiet()`. The `xlim` should be `wlim`.